replicate/configmap/configmaps.go

```diff
@@ -276,6 +276,29 @@ func (r *Replicator) DeleteReplicatedResource(targetResource interface{}) error
 			return errors.Wrapf(err, "Failed deleting %s: %v", targetLocation, err)
 		}
 	} else {
+		var patch []common.JSONPatchOperation
+		exists := make(map[string]struct{})
+		for _, value := range resourceKeys {
+			exists[value] = struct{}{}
+		}
+		for _, val := range strings.Split(object.Annotations[common.ReplicatedKeysAnnotation], ",") {
+			if _, ok := exists[val]; ok {
+				patch = append(patch, common.JSONPatchOperation{Operation: "remove", Path: fmt.Sprintf("/data/%s", val)})
+			}
+		}
+		patch = append(patch, common.JSONPatchOperation{Operation: "remove", Path: fmt.Sprintf("/metadata/annotations/%s", strings.Replace(common.ReplicatedKeysAnnotation, "/", "~1", 1))})
```

Comment:
I think it would be useful to move this to a common method like `common.JSONPatchPathEscape()`. This would require the `Replace` to handle all occurrences of a given character.

Reply:
Added the function in common.go.
Found an issue when a resource inherits from multiple sources, since the entire annotation gets deleted, so I'll rework it to update the annotation instead